great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -195,11 +205,25 @@ def _generate_batch_spec_parameters_from_batch_definition(
     def build_batch_spec(
         self,
         batch_definition: BatchDefinition,
-        batch_data: Any,
-    ) -> RuntimeDataBatchSpec:
-        batch_spec = super().build_batch_spec(batch_definition=batch_definition)
-        batch_spec["batch_data"] = batch_data
-        return RuntimeDataBatchSpec(batch_spec)
+        runtime_parameters: dict,
+    ) -> Union[RuntimeDataBatchSpec, RuntimeQueryBatchSpec, PathBatchSpec]:
+        batch_spec: BatchSpec = super().build_batch_spec(
+            batch_definition=batch_definition
+        )
+        if runtime_parameters.get("batch_data") is not None:
+            batch_spec["batch_data"] = runtime_parameters.get("batch_data")
+            return RuntimeDataBatchSpec(batch_spec)
+        elif runtime_parameters.get("query"):
+            batch_spec["query"] = runtime_parameters.get("query")
+            return RuntimeQueryBatchSpec(batch_spec)
+        elif runtime_parameters.get("path"):
+            path = runtime_parameters.get("path")
+            batch_spec["path"] = path
+            parsed_url = urlparse(path)
+            if "s3" in parsed_url.scheme:
+                return S3BatchSpec(batch_spec)
+            else:
+                return PathBatchSpec(batch_spec)
```

Comment:
Would it make sense to have an `else:` here where we return `None` or raise an error, or is this sufficiently handled with `_validate_runtime_parameters()` ?